pkg/network/server.go

```diff
@@ -825,14 +823,14 @@ func (s *Server) requestBlocksOrHeaders(p Peer) error {
 		return nil
 	}
 	var (
-		bq              bqueue.Blockqueuer = s.chain
+		bq              bqueue.Blockqueuer[*block.Block] = chainBlockQueueAdapter{s.chain}
```

Comment:
... and reuse this interface here instead of `bqueue.Blockqueuer` because `bqueue.Blockqueuer` is a queue-specific thing now.